src/gt4sd/algorithms/generation/tests/test_diffusion.py

```diff
@@ -242,3 +255,28 @@ def test_generation_via_registry(
     assert len(samples) == 1
     sample = samples[0]
     assert isinstance(sample, PIL.Image.Image)
+
+
+@pytest.mark.parametrize(
+    "config, algorithm",
+    [
+        pytest.param(
+            GeoDiffGenerator,
+            DiffusersGenerationAlgorithm,
+        ),
+    ],
+)
+def test_geodiff_conditional_generation_via_import(config, algorithm):
+
+    with importlib_resources.as_file(
+        importlib_resources.files("gt4sd") / "algorithms/generation/tests/mol_dct.pkl"
+    ) as path:
+        with open(path, "rb") as f:
+            loaded_dict = pickle.load(f)
+        prompt = Data.from_dict(loaded_dict[0])
+        configuration = config(prompt=prompt)
```

Comment:
let's ensure the prompt is a dictionary and create the data inside the implementation, the dictionary should be easy to serialise/deserialise